### PR TITLE
feat(react-tags): Replace `avatar` prop with `media`; polish styles for spacing

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
This PR continues react-tags implementation.

Important changes:
- Tag/TagButton `avatar: Slot<Avatar>` prop is replaced with `media: Slot<span>`. `AvatarContext` is used to provide default size/shape to avatar.
- Tag/TagButton renders children as primary text: `<Tag>Primary text</Tag>` renders image
- Styles for medium sized Tag layout:
    - Tag  ([figma](https://www.figma.com/file/0JIwHyPDlwQXbssABMSd3h/Tag-(previously-Pill)?node-id=6881-105095&t=ZA6YOo2ccbXGoyIE-4)): 
        Screenshot 2023-04-28 at 12 59 24
    - TagButton ([figma](https://www.figma.com/file/0JIwHyPDlwQXbssABMSd3h/Tag-(previously-Pill)?type=design&node-id=6881-114771&t=vBTYbYevi1IoABR7-4)): 
        Screenshot 2023-04-28 at 12 59 34

TODOs for follow up PRs:
- Styles for small/extra-small size. Styles for Tag colors. Styles for outlined Tag and Tag in hover/pressed/selected state.
- Tag behavior: dismiss and select Tag

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/4acd563e14d4a257585cff2dba2fb826ce9c059b'>4acd563e14d4a257585cff2dba2fb826ce9c059b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/Avatar-mount_base.html">716</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/Avatar-mount.html">706</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/Button-mount_base.html">394</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/Button-mount.html">401</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/Field-mount_base.html">1225</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/Field-mount.html">1204</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/FluentProvider-mount_base.html">778</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/FluentProvider-mount.html">766</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/FluentProviderWithTheme-mount_base.html">114</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/FluentProviderWithTheme-mount.html">109</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">91</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">98</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">109</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">98</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/InfoButton-mount_base.html">20</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/InfoButton-mount.html">19</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/MakeStyles-mount_base.html">1211</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/MakeStyles-mount.html">1181</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/Persona-mount_base.html">1934</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/Persona-mount.html">1887</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/SpinButton-mount_base.html">1431</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27725/perf-test-react-components/SpinButton-mount.html">1472</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: apps/public-docsite-v9/.storybook/main.js [hunk 1]
```diff
@@ -15,6 +15,8 @@ module.exports = /** @type {Omit<import('../../../.storybook/main'), 'typescript
     '../../../packages/react-components/react-migration-v0-v9/stories/**/@(index.stories.@(ts|tsx)|*.stories.mdx)',
     '../../../packages/react-components/react-migration-v8-v9/stories/**/@(index.stories.@(ts|tsx)|*.stories.mdx)',
     '../../../packages/react-components/react-datepicker-compat/stories/**/@(index.stories.@(ts|tsx)|*.stories.mdx)',
+    // TODO: remove this before PR merge. Just for the purpose of building PR docsite
+    '../../../packages/react-components/react-tags/stories/**/@(index.stories.@(ts|tsx)|*.stories.mdx)',
```
Comment (YuanboXue-Amber) [inline apps/public-docsite-v9/.storybook/main.js:19]: a comment to remind myself to remove this

File: packages/react-components/react-avatar/src/components/Avatar/Avatar.types.ts [hunk 2]
```diff
@@ -11,6 +11,11 @@ export type AvatarSizes = AvatarSize;
  */
 export type AvatarSize = 16 | 20 | 24 | 28 | 32 | 36 | 40 | 48 | 56 | 64 | 72 | 96 | 120 | 128;
 
+/**
+ * Shape of the avatar
+ */
+export type AvatarShape = 'circular' | 'square';
+
```
Comment (sopranopillow) [inline packages/react-components/react-avatar/src/components/Avatar/Avatar.types.ts:18]: Would it be possible to split the Avatar related changes into another PR?

File: packages/react-components/react-tags/src/components/Tag/useTag.tsx [hunk 3]
```diff
@@ -43,17 +56,20 @@ export const useTag_unstable = (props: TagProps, ref: React.Ref<HTMLElement>): T
...
-    avatar: resolveShorthand(props.avatar),
+    avatarSize: tagAvatarSizeMap[size],
+    avatarShape: tagAvatarShapeMap[shape],
+    media: resolveShorthand(props.media),
+
+    content: resolveShorthand(props.content, { required: !!props.primaryText || !!props.children }),
     icon: resolveShorthand(props.icon),
-    primaryText: resolveShorthand(props.primaryText),
+    primaryText: resolveShorthand(props.primaryText, { required: true }),
     secondaryText: resolveShorthand(props.secondaryText),
```
Comment (sopranopillow) [inline packages/react-components/react-tags/src/components/Tag/useTag.tsx:66]: Also would be nice to have separate PRs for the changes in TagButton and Tag.

File: packages/react-components/react-tags/src/components/Tag/Tag.types.ts [hunk 4]
```diff
@@ -1,20 +1,46 @@
...
+   * Slot for an icon or other visual element
+   */
+  media: Slot<'span'>;
+
+  /**
+   * A layout wrapper for the icon slot, the primaryText and secondaryText slots
+   */
+  content: Slot<'div'>;
+
+  icon: Slot<'span'>;
```
Comment (ling1726) [inline packages/react-components/react-tags/src/components/Tag/Tag.types.ts:25]: There's both a media slot whose description is 'slot for an icon or other visual element' and also an icon slot

is there any way we could make this naming a bit clearer?

File: packages/react-components/react-tags/src/components/Tag/Tag.types.ts [hunk 5]
```diff
@@ -1,20 +1,46 @@
...
+
+  /**
+   * Main text for the Tag. Children of the root slot are automatically rendered here
+   */
+  primaryText: Slot<'span'>;
+
+  /**
+   * Secondary text that describes or complements the main text
+   */
+  secondaryText: Slot<'span'>;
```
Comment (ling1726) [inline packages/react-components/react-tags/src/components/Tag/Tag.types.ts:35]: instead of `secondaryText` maybe you could use something like in table https://github.com/microsoft/fluentui/blob/e9c67ecb4f3b0ee1ee355e2c8eec17afa0b20a5a/packages/react-components/react-table/src/components/TableCellLayout/TableCellLayout.types.ts#L24-L27

Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/Tag.types.ts:35]: :D I was actually coping exactly this component -> how did you know 😱

Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/Tag.types.ts:35]: Learnt how to do pass children into primaryText(main) slot from it :D

File: packages/react-components/react-tags/src/components/Tag/useTag.tsx [hunk 6]
```diff
@@ -43,17 +56,20 @@ export const useTag_unstable = (props: TagProps, ref: React.Ref<HTMLElement>): T
       ref,
       ...props,
     }),
-    content: resolveShorthand(props.content, { required: true }),
-    avatar: resolveShorthand(props.avatar),
+    avatarSize: tagAvatarSizeMap[size],
+    avatarShape: tagAvatarShapeMap[shape],
+    media: resolveShorthand(props.media),
+
+    content: resolveShorthand(props.content, { required: !!props.primaryText || !!props.children }),
```
Comment (ling1726) [inline packages/react-components/react-tags/src/components/Tag/useTag.tsx:63]: if `primaryText` slot is always required, then shouldn't `content` be required?

File: packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx [hunk 7]
```diff
@@ -14,47 +13,12 @@ import { Avatar } from '@fluentui/react-avatar';
...
-  return {
-    components: {
-      root: 'div',
-      contentButton: 'button',
-      avatar: Avatar,
-      icon: 'span',
-      primaryText: 'span',
-      secondaryText: 'span',
-      dismissButton: 'button',
+  const state = useTag_unstable(props, ref);
```
Comment (ling1726) [inline packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx:16]: Couldn't you use do

```ts
useTag_unstable({ tabIndex: 0, ...props }, ref)
```

File: packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+    gridTemplateAreas: `
+    "media icon .        "
+    "media icon primary  "
+    "media icon secondary"
+    "media icon .        "
+    `,
+    paddingRight: tokens.spacingHorizontalS,
+
+    position: 'relative',
+    ...createCustomFocusIndicatorStyle({
```
[the remark this answers is not in the document]
Comment (ling1726) [reply packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts:53]: what else is different about the tag button styles and tag styles?

File: packages/react-components/react-tags/src/components/Tag/__snapshots__/Tag.test.tsx.snap [hunk 9]
```diff
@@ -5,21 +5,13 @@ exports[`Tag renders a default state 1`] = `
   <div
     class="fui-Tag"
   >
-    <span
+    <div
```
Comment (spmonahan) [inline packages/react-components/react-tags/src/components/Tag/__snapshots__/Tag.test.tsx.snap:8]: Can we remove the snapshot test in favor of a simple "it renders correctly" test that asserts the relevant parts of a simple, default render case? Or maybe just remove the test altogether in this case since the default `Tag` render is simple and conformance tests should cover the important parts like having the correct classes applied.

Examples:
- [Button](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-button/src/components/Button/Button.test.tsx#L36)
- [Field](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-field/src/components/Field/Field.test.tsx#L37)

File: packages/react-components/react-tags/src/components/Tag/Tag.types.ts [hunk 10]
```diff
@@ -1,20 +1,46 @@
...
+
+  dismissButton: Slot<typeof Button>;
```
Comment (spmonahan) [inline packages/react-components/react-tags/src/components/Tag/Tag.types.ts:37]: One of the reasons to eliminate the `Avatar` slot in favor of media is to reduce the overall bundle size of `Tag`. How do we benefit by using Fluent `Button` here as it will also increase the bundle size?

File: packages/react-components/react-tags/src/components/Tag/useTagStyles.styles.ts [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+  primaryText: 'fui-Tag__primaryText',
+  secondaryText: 'fui-Tag__secondaryText',
+  dismissButton: 'fui-Tag__dismissButton',
+};
+
+/**
+ * Styles for the root slot
+ */
+const useStyles = makeStyles({
+  root: {
```
Comment (spmonahan) [inline packages/react-components/react-tags/src/components/Tag/useTagStyles.styles.ts:20]: This can be deferred to a later PR but consider using `makeResetStyles()` for the base styles as this greatly reduces the default number of classes on a component which improves style-related perf.

[Example](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.styles.ts#L22)

File: packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx (hunk 7 above)
Comment (spmonahan) [reply packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx:16]: I think it would need to be 
```ts
useTag_unstable({ ...props, content: { tabIndex: 0, ...props.content } }, ref)
``` 
or make `content` the primary slot.

Comment (spmonahan) [reply packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx:16]: Can you remind me what the distinction between `Tag` and `TagButton` is?

If it's just the `tabIndex` maybe we should look at a way to implement this behavior as a prop on `Tag`?

File: packages/react-components/react-tags/stories/Tag/TagDefault.stories.tsx [hunk 12]
```diff
@@ -1,20 +1,148 @@
...
-        src: 'https://fabricweb.azureedge.net/fabric-website/assets/images/avatar/KatriAthokas.jpg',
-      },
-    }}
-    icon={<Calendar3Day28Regular />}
-    primaryText="Primary text"
-    secondaryText="Secondary text"
-    dismissable={true}
-    {...props}
-  />
+  <div style={{ display: 'flex', columnGap: 28 }}>
```
Comment (spmonahan) [inline packages/react-components/react-tags/stories/Tag/TagDefault.stories.tsx:8]: Nit: use `makeStyles` for the examples as that's best practice.

File: packages/react-components/react-tags/src/components/Tag/Tag.types.ts (hunk 4 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/Tag.types.ts:25]: The only reason I have icon as a separate slot is that I can't figure out a style for one slot. Avatar has 8px gap to the text while icon have 2px:
Screenshot 2023-05-01 at 20 33 32

File: packages/react-components/react-tags/src/components/Tag/Tag.types.ts (hunk 5 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/Tag.types.ts:35]: I kept the `primary text/secondary text` as in the previous PR. 
In design spec, I found there's both `primary content/secondary content` and `primary text/secondary text` :D Personally I prefer 'text' over 'content' as we have a 'content' slot, and it's kind of confusing to have another 'primary content'. But I'm open for changing this. I'll chat with the designer on the naming.

File: packages/react-components/react-tags/src/components/Tag/useTag.tsx (hunk 6 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/useTag.tsx:63]: good catch, let me update this

File: packages/react-components/react-tags/src/components/Tag/useTagStyles.styles.ts [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+    display: 'inline-flex',
+    height: '32px',
+    width: 'fit-content',
+    ...shorthands.borderRadius(tokens.borderRadiusMedium),
+
+    backgroundColor: tokens.colorNeutralBackground3,
```
[the remark this answers is not in the document]
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/useTagStyles.styles.ts:26]: Yeah I follow the spec. If this is weird, should I raise a question to designer?

File: packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx (hunk 7 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx:16]: I'm having doubts on TagButton as well. 
- TagButton looks and works really similar as a split button: it has two focusable parts, the content and the 'x' icon. While Tag has only the 'x' icon focusable. But this can be done by an extra prop on Tag. 
- there're style differences, TagButton has an extra divider and different focus rings, but this is nothing that can't be solved by style override.
- The deal breaker for me - Tag doesn't have select state but TagButton can. This makes me think TagButton as a separate component can be cleaner. 
When I started working on Tag, there's already a TagButton component so I figure it would be easier to keep it as is. And maybe merge these two components as we see how things goes. It feels easier than creating one component and split it when we must.

File: packages/react-components/react-avatar/src/components/Avatar/Avatar.types.ts (hunk 2 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-avatar/src/components/Avatar/Avatar.types.ts:18]: I created #27738 for avatar shape

File: packages/react-components/react-tags/stories/Tag/TagDefault.stories.tsx (hunk 12 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/stories/Tag/TagDefault.stories.tsx:8]: I updated all stories to use makeStyles except for the default stories. The default stories now contain may tags just so I have an easier time working on them. I left a TODO there to eventually have only one tag/tagButton in the default stories.

File: packages/react-components/react-tags/src/components/Tag/__snapshots__/Tag.test.tsx.snap (hunk 9 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/__snapshots__/Tag.test.tsx.snap]: I removed the snapshot test altogether :) I think conformance tests are good enough for now

File: packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx (hunk 7 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx:16]: And also content doesn't like to be spreaded :/ :
image

File: packages/react-components/react-tags/src/components/Tag/useTag.tsx (hunk 6 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/useTag.tsx]: Done :)

File: packages/react-components/react-tags/src/components/Tag/useTagStyles.styles.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (ling1726) [reply packages/react-components/react-tags/src/components/Tag/useTagStyles.styles.ts:26]: yes please raise this with the designer. AFAIK is using the token sets incorrectly

File: packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts:24]: Based on our call, I exported `useTagBaseStyle` from _useTagStyles.styles.ts_, and used it in __useTagButtonStyles.styles.ts_. Can you check if I got your point correctly?

File: packages/react-components/react-tags/src/components/Tag/Tag.types.ts (hunk 10 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/Tag.types.ts]: Good point :) I think there's no reason to use Button here. I updated it to use `Slot<'button'>`.

File: packages/react-components/react-tags/src/components/Tag/useTagStyles.styles.ts (hunk 11 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/useTagStyles.styles.ts:21]: Agree :) I added a TODO in the comment for future PRs

File: packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx (hunk 7 above)
Comment (spmonahan) [reply packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx:16]: Maybe it's just a typing issue? Anythings @behowell ?

Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/TagButton/useTagButton.tsx]: Yeah this works :) updated

File: packages/react-components/react-tags/src/components/Tag/useTagStyles.styles.ts [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+  },
+  secondaryText: {
+    ...shorthands.gridArea('secondary'),
+    paddingLeft: tokens.spacingHorizontalXXS,
+    paddingRight: tokens.spacingHorizontalXXS,
+    ...typographyStyles.caption2,
+  },
+
+  resetButton: {
+    boxSizing: 'content-box',
```
[the remark this answers is not in the document]
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/useTagStyles.styles.ts:85]: `button` has box-sizing as `border-box`. This style resets the default button style, so it has this

File: packages/react-components/react-tags/src/components/Tag/Tag.types.ts (hunk 4 above)
Comment (ling1726) [reply packages/react-components/react-tags/src/components/Tag/Tag.types.ts:24]: Oh then I completely misunderstood the purpose of this slot: we should not have it . if someone puts an icon in the media slot... they have broken all design 🥳🥳🥳

File: packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (ling1726) [reply packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts:24]: @YuanboXue-Amber  please double check these focus indicator styles like we did for Tag offline

File: packages/react-components/react-tags/src/components/Tag/useTagStyles.styles.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/useTagStyles.styles.ts:28]: I confirmed with design it is intentional :D

File: packages/react-components/react-tags/src/components/Tag/Tag.types.ts (hunk 4 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/Tag.types.ts:24]: I brought this up with design, Toshie will try to figure out if there's a solution to make avatar and icon fit in one slot. In the meantime, I'll use bernardo's idea and fit them into one gridArea :)

File: packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts]: Double checked :) and fixed the TagButton focus line the same way

[the remark this answers is not in the document]
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts]: Btw @ling1726 I found out I need to apply `{enableOutline: true}` for each `createCustomFocusIndicatorStyle`. 
For example I have a base focus line for content, and a overriding one with sharp border radius when there's dismiss button. I will have to apply `{enableOutline: true}` on the overriding one as well. It makes sense, but it also doesn't feel very nice when I have `{enableOutline: true}` everywhere.

File: packages/react-components/react-tags/src/components/Tag/Tag.types.ts (hunk 4 above)
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/Tag/Tag.types.ts:24]: > I'll start by saying I'm not sure if this is a good idea :)
> 
> What do you think about defaulting to the icon spacing and then using the Avatar context to override the spacing for Avatars?

@spmonahan I don't like two slots either :D But I don't have a good way of doing style overrides. I chat with Designer on this issue to see if we can adjust the design to fit both avatar/icon in one slot.
I've tried having `className` on AvatarContext (#27755) and use it to apply style override for avatar on top of icon slot. But it was point out to me that AvatarContext is not supposed to be used this way. TBH it feels weird to me as well.

File: packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (YuanboXue-Amber) [reply packages/react-components/react-tags/src/components/TagButton/useTagButtonStyles.styles.ts]: Yep I merged master and it works! I removed all unnecessary  `{enableOutline: true}`
